active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+- These APIs use different code paths depending on whether WinUI 3 is running on Win32 or UWP. 
+- There are some APIs that don’t work on Win32 or UWP yet in this phase. Those APIs could return **null** or **throw** an exception depending on the case. 
+
+This agreement applies to the Application class too.
+
+## Window Class
+Namespace: Microsoft.UI.Xaml
+
+```CS
+public class Window : IClosable
+```
+
+### **Window management, appearance, and behaviors**
+
+### Window() constructor **[NEW]** 
+The constructor initializes the Window object. Windows are created in the same thread where they are instantiated. Frequently, this thread is the UI Thread.
+
+```CS
+public Window();
+```
+#### Samples
+##### Code Sample
+```CS
+Window window = new Window();
+window.Content = new TextBlock() { Text = "Hello" };
+window.Activate(); 
+```
+##### Markup Sample
+
+```XML
+<Window 
+    x:Class="MainWindow"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+>
+    <Grid><Button Content=”Click” Click=”OnClick”/></Grid>
+</Window>
+```
+```CS
+public partial class MainWindow:Window
+{
+  public MainWindow()
+  {
+	InitializeComponent();
+  }
+}
+```
+##### Multiple Widndows Sample
+Developers can create multi-window experiences with WinUI 3 in the same UI thread. 
+```CS
+Window w = new Window ();
+Button b = new Button() { Content = "Click Me" };
+b.Click += OnClick;
+w.Content = b;
+w.Activate();
+```
+```CS
+private void OnClick(object sender, RoutedEventArgs e)
+{
+   Window w2 = new Window ();
+   w2.Content = new TextBlock(){Text=”Window 2”}
+   w2.Activate();
+}
+```
+##### Create a Window in a separated thread
+To create a Window in a separated thread, you should create the thread first and create a Window in this new Thread. 
+```CS
+private void Button_Click(object sender, RoutedEventArgs e)
+{
+    Thread thread = new Thread(() =>
+    {
+        Window w = new Window();
+        w.Content = new TextBlock() { Text = "Separated Thread Window" };
+        w.Activate();
+        w.Closed += (sender1, e1) => w.Dispatcher.InvokeShutdown();
+
+        System.Windows.Threading.Dispatcher.Run();
+
+    });
+    thread.SetApartmentState(ApartmentState.STA);
+    thread.IsBackground = true; 
+    thread.Start();
+} 
+```
+
+ #### Remarks
+
+- **Win32**: The constructor creates a top-level Win32 Window (HWnd). A top-level Window is a window that is not a child window or has no parent window. 
+
+- **UWP**: There is already one CoreWindow created and UWP only allows one CoreWindow per thread, as a consequence, the contructor and the XAML Markup throw an exception when running in UWP.
+    - Exception type: TBD
+ 
+    - Exception Message: TBD. Something like "Only one Window per thread is allowed in UWP app model".
+
+### Window.Bounds property
+Gets a Rect value containing the height and width of the application window in logical pixels.
+
+```CS
+public Rect Bounds { get; }
```

Comment:
>Bounds [](start = 12, length = 6)

A "Rect" is particularly problematic. Can we change this to ClientSize or something? We'll want to distinguish from FrameSize, which is in a different coordinate space and affected by borders, etc.

Reply:
This is an existing API. But see the [other comment](https://github.com/microsoft/microsoft-ui-xaml-specs/pull/71/files/9c72d8dde073924516b125e1ddaab1de36c61bc4#r393373967) about this property.